### PR TITLE
[WIP] [feat] Tasks CRUD functionalities

### PR DESCRIPTION
**Pull request by shivampandey0**
- Add Task
- Edit Task
- Delete Task
- localStorage persistance
- Data Context

image

### CODE REVIEW

File: src/context/DataContext.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useState, useEffect } from "react";
+
+const DataContext = createContext();
+
+const useData = () => useContext(DataContext);
+
+const DataProvider = ({ children }) => {
+  const [taskList, setTaskList] = useState(
+    JSON.parse(localStorage.getItem("tasks")) ?? []
```
Comment (Hamdrive) [inline src/context/DataContext.jsx:9]: Nice use of nullish coalescing operator for handling undefined return from JSON.parse

**Review by Hamdrive — COMMENTED** [review]
Great implementation of context to handle task functionalities, 
liked the use of nullish coalescing operator even though pipe operator would have done the same 
Overall good implementation of logic and clean code

Suggestion to look into alternate color scheme, as this seems a bit too strong to the eyes

**Review by junaidshaikh-js — APPROVED** [review]
Good folder structure. 

Nice that you took care of accessibility by adding aria-labels to input and text area elements. Can be merged.

File: src/context/DataContext.jsx (hunk 1 above)
Comment (shivampandey0) [reply src/context/DataContext.jsx:9]: Thanks for the suggestion will change the colors in the end while doing style optimizations.

**Pull request merged**